### PR TITLE
lua_api.txt: Various fixes and improvements

### LINKED ISSUE #6106: minetest.registered_aliases is not documented
The existance of `minetest.registered_aliases` is not documented in `lua_api.txt`

### LINKED ISSUE #7783: not_in_creative_inventory group is not documented (remove 'soil' from engine docs)
##### Issue type

 - Documentation issue

##### Summary

`not_in_creative_inventory` is missing from the lua_api.txt. Should probably be listed under [Special Groups](https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L1526-L1553). Not that it is a huge problem, but I got really confused when I was searching for it to make sure I got the name right...

### PR DESCRIPTION
**Pull request by pauloue**
- Added missing tables of registered things
    - Fixes #6106 and checks a box in #5854
    - Used a catchall statement for tables of registered callbacks since they aren't likely to be used much and there's a lot.
- Closes #7783 
- More stuff, see commits

### CODE REVIEW

**Comment by ClobberXD** [comment]
IIRC, the reason for not including all `registered_*` tables in the documentation is because mods aren't supposed to access all of them.

**Comment by pauloue** [comment]
Hm yes the registered callback functions probably should not be documented.

File: doc/lua_api.txt [hunk 1]
```diff
@@ -3560,15 +3508,48 @@ Call these functions only at load time!
...
     * Example: `minetest.override_item("default:mese", {light_source=LIGHT_MAX})`
 * `minetest.clear_registered_ores()`
+    * Clears all ores currently registered.
 * `minetest.clear_registered_biomes()`
+    * Clears all biomes currently registered.
 * `minetest.clear_registered_decorations()`
+    * Clears all decorations currently registered.
+* `minetest.clear_registered_schematics()`
+    * Clears all schematics currently registered.
+* `minetest.register_chatcommand(cmd, chatcommand definition)`
```
Comment (SmallJoker) [inline doc/lua_api.txt]: I'd prefer to have these functions somehow grouped
Suggestion:
1) `Gameplay registration functions`: chat, privilege, auth
2) `Environment registration functions`: items, biome, craft

Comment (pauloue) [reply doc/lua_api.txt]: See latest commit

**Pull request merged**